apps/src/applab/GoogleChart.js

```diff
@@ -0,0 +1,463 @@
+ *          rawData, (hopefully) in the order they were defined in the row
+ *          objects.
+ */
+GoogleChart.inferColumnsFromRawData = function (rawData) {
+  return Object.getOwnPropertyNames(rawData.reduce(function (memo, row) {
+    Object.getOwnPropertyNames(row).forEach(function (key) {
+      memo[key] = true;
+    });
+    return memo;
+  }, {}));
+};
+
+/**
+ * Makes sure data looks okay, throws errors and logs warnings as appropriate.
+ * @param {string[]} columns
+ * @param {Object[]} data
+ * @private
+ */
+GoogleChart.prototype.verifyData_ = function (data, columns) {
+  // Warn when no rows are present
+  if (data.length === 0) {
+    this.warn('No data.');
+  }
+
+  // Error when not enough columns are provided
+  if (columns.length < 2) {
+    throw new Error('Not enough columns for chart; expected at least 2.');
+  }
+
+  // Warn on empty columns?
+  columns.forEach(function (colName) {
+    var exists = data.some(function (row) {
+      return row[colName] !== undefined;
+    });
+    if (!exists) {
+      this.warn('No data found for column "' + colName + '".');
+    }
+  }.bind(this));
+};
+
+/**
+ * @param {Object[]} rows - Rows as POJOs with keys.
+ * @param {string[]} columns - Column names which must correspond to keys
+ *        in the row objects.
+ * @return {google.visualization.DataTable}
+ */
+GoogleChart.dataTableFromRowsAndColumns = function (rows, columns) {
+  var dataArray = rows.map(function (row) {
+    return columns.map(function (key) {
+      return row[key];
+    });
+  });
+  return GoogleChart.lib.visualization.arrayToDataTable([columns].concat(dataArray));
+};
+
+/* jshint unused: false */
+/**
+ * Internal 'abstract' method that subclasses should use to implement the actual
+ * rendering step.
+ *
+ * @param {google.visualzation.DataTable} dataTable
+ * @param {Object} options
+ * @returns {Promise}
+ * @private
+ */
+GoogleChart.prototype.render_ = function (dataTable, options) {
+  return Promise.reject(new Error('Rendering unimplemented for chart type.'));
+};
+/* jshint unused: true */
+
+/**
+ * Google Charts API Pie Chart
+ *
+ * @see https://developers.google.com/chart/interactive/docs/gallery/piechart
+ *
+ * @param {Element} targetDiv
+ * @constructor
+ * @extends GoogleChart
+ */
+var PieChart = function (targetDiv) {
+  GoogleChart.call(this, targetDiv);
+};
+PieChart.inherits(GoogleChart);
+GoogleChart.PieChart = PieChart;
+
+
+PieChart.prototype.render_ = function (dataTable, options) {
+  var apiChart = new GoogleChart.lib.visualization.PieChart(this.targetDiv_);
+  apiChart.draw(dataTable, options);
+  return Promise.resolve();
+};
+
+/**
+ *
+ * @param {string[]} columns
+ * @param {Object[]} data
+ * @private
+ * @override
+ */
+PieChart.prototype.verifyData_ = function (data, columns) {
+  PieChart.superPrototype.verifyData_.call(this, data, columns);
+
+  if (columns.length > 2) {
+    this.warn('Too many columns for pie chart; only using the first 2.');
```

Comment:
You are correct! I based this off of my own comment in the spec which I see now is wrong.